ui/design-system-core/src/components/Icon/index.tsx

```diff
@@ -34,46 +39,91 @@ export const iconTypes: IconType[] = [
   'twitter',
 ];
 
-const fillIcons: IconType[] = ['akasha', 'appModeration'];
+/*@TODO: change all custom icons into ones in the design system and remove the following */
+const fillOnlyIcons: IconType[] = ['akasha', 'appModeration'];
+
+/*@TODO: change all custom icons into ones in the design system and remove the following */
+const fillAndStrokeIcons: IconType[] = ['twitter', 'telegram', 'discord'];
 
 const Icon: React.FC<IconProps> = props => {
   const {
     type,
     ref,
-    isCustomIcon,
     plain,
     accentColor,
     clickable,
+    size = 'md',
+    breakPointSize,
     color,
     disabled,
     testId,
     styling,
   } = props;
 
-  const PassedIcon = isCustomIcon || !HeroIcons[type] ? CustomIcons[type] : HeroIcons[type];
+  const breakPointStyle = breakPointSize
+    ? ICON_SIZE_MAP_BY_BREAKPOINT(breakPointSize.breakPoint)[breakPointSize.size]
+    : '';
+
+  const sizeStyle =
+    typeof size === 'object'
+      ? `${getWidthClasses(size?.width)} ${getHeightClasses(size?.height)}`
+      : `${ICON_SIZE_MAP[size]} ${breakPointStyle}`;
+
+  const PassedIcon = !HeroIcons[type] ? CustomIcons[type] : HeroIcons[type];
 
   if (!PassedIcon) {
     // tslint:disable-next-line no-console
     console.error('There is no such icon', type);
     return null;
   }
 
-  const isfill = fillIcons.includes(type as IconType);
+  /*@TODO: change all custom icons into ones in the design system and remove the following */
+  const isFillOnlyIcon = fillOnlyIcons.includes(type);
 
-  // this determines what prop to target - 'fill' for fillIcons, 'stroke' for others
-  const svgPrefix = isfill ? 'fill' : 'stroke';
+  /*@TODO: change all custom icons into ones in the design system and remove the following */
+  const isFillAndStrokeIcon = fillAndStrokeIcons.includes(type);
 
-  const className = `flex items-center justify-center select-none ${
-    plain ? `[& *]:${svgPrefix}-black dark:[& *]:${svgPrefix}-white` : ''
-  } ${color ? `[& *]:${svgPrefix}-${color}` : ''} ${
-    accentColor ? `[& *]:${svgPrefix}-secondary-dark` : ''
-  } ${clickable && !disabled ? `cursor-pointer hover:[& *]:${svgPrefix}-secondary-dark` : ''}`;
+  const plainStyle = plain
+    ? `${isFillOnlyIcon ? '' : '[&>*]:stroke-black dark:[&>*]:stroke-white'} ${
+        isFillAndStrokeIcon ? '[&>*]:fill-transparent' : ''
+      }`
+    : '';
+  const colorStyle = color ? `[&>*]:stroke-${color} [&>*]:fill-${color}` : '';
+  const accentColorStyle = accentColor
+    ? `${isFillOnlyIcon ? '' : '[&>*]:stroke-secondary-light dark:[&>*]:stroke-secondary-dark'} ${
+        isFillAndStrokeIcon ? '[&>*]:fill-transparent' : ''
+      }`
+    : '';
```

Comment:
and `accentColor` sets the corresponding secondary colors.

Reply:
This part will become leaner when we replace the existing custom icons with the once on the design system. Then we won't need to check for things like `isFillOnlyIcon` and `isFillAndStrokeIcon`. I will have this changes on my next PR.